cloudsync/sync/state.py

```diff
@@ -1168,6 +1168,21 @@ def change(self, age):
         if not change_set:  # todo: cache the changeset
             return None
 
+        # fill in path if needed. Note that this updates the priority of the SyncEntry
+        for e in change_set:
+            for side in (LOCAL, REMOTE):
+                oid = e[side].oid
+                if oid and not e[side].path and e[side].exists in (EXISTS, UNKNOWN):
+                    info = self.providers[side].info_oid(oid, use_cache=False)
+                    e[side]._last_gotten = time.time()
+                    if info:
+                        self.update_entry(
+                            e, side, oid, path=info.path, file_hash=info.hash, exists=True,
+                            otype=info.otype, size=info.size, mtime=info.mtime
+                        )
+                    else:
+                        self.update_entry(e, side, oid, exists=False)
```

Comment:
This can be replace with a single call to `unconditionally_get_latest` if you add an optional argument `changed=True` to make changing the change time optional, and pass in `False` here

Reply:
updated to reuse `uncoditionally_get_latest()`